File: app/src/main/java/com/ramez/shopp/Adapter/BrandsAdapter.java
package com.ramez.shopp.Adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;

import com.ramez.shopp.Classes.CategoryModel;
import com.ramez.shopp.Classes.Constants;
import com.ramez.shopp.Classes.GlobalData;
import com.ramez.shopp.Classes.UtilityApp;
import com.ramez.shopp.Models.BrandModel;
import com.ramez.shopp.R;
import com.ramez.shopp.databinding.RowBandItemBinding;

import java.util.List;

public class BrandsAdapter extends RecyclerView.Adapter<BrandsAdapter.Holder> {

    private Context context;
    private List<BrandModel> list;
    private OnBrandClick onBrandClick;
    private int limit;

    public BrandsAdapter(Context context, List<BrandModel> categoryDMS, OnBrandClick onBrandClick,int limit) {
        this.context = context;
        this.list = categoryDMS;
        this.onBrandClick = onBrandClick;
        this.limit=limit;
        ;
    }

    @Override
    public Holder onCreateViewHolder(ViewGroup parent, int viewType) {

        RowBandItemBinding itemView = RowBandItemBinding.inflate(LayoutInflater.from(context), parent, false);

        return new Holder(itemView);
    }


    @Override
    public void onBindViewHolder(final Holder holder, int position) {
        BrandModel brandModel=list.get(position);
        String imageUrl="";

        if (UtilityApp.getLanguage().equals(Constants.English)) {
            imageUrl=brandModel.getImage();

        }
        else {
            imageUrl=brandModel.getImage2();

        }

        GlobalData.PicassoImg(imageUrl
                ,R.drawable.holder_image,holder.binding.ivCatImage);


        holder.binding.container.setOnClickListener(v ->
        {
            onBrandClick.onBrandClicked(position,brandModel);
        });
    }

    public void setCategoriesList(List<BrandModel> list) {
        this.list = list;
        notifyDataSetChanged();
    }

    @Override
    public int getItemCount() {
        if (limit != 0) return Math.min(list.size(), limit);
        else return list.size();
    }

    public interface OnBrandClick {
        void onBrandClicked(int position, BrandModel brandModel);
    }

    static class Holder extends RecyclerView.ViewHolder {

        RowBandItemBinding binding;

        Holder(RowBandItemBinding view) {
            super(view.getRoot());
            binding = view;
        }
    }
}

File: app/src/main/java/com/amour/shop/adapter/BrandsAdapter.kt
package com.amour.shop.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.amour.shop.Models.BrandModel
import com.amour.shop.R
import com.amour.shop.classes.Constants
import com.amour.shop.classes.GlobalData.GlideImg
import com.amour.shop.classes.UtilityApp
import com.amour.shop.databinding.RowBandItemBinding


 class BrandsAdapter(
    private val context: Context,
    private var list: MutableList<BrandModel>?,
    private val onBrandClick: OnBrandClick,
    private val limit: Int
) :
        RecyclerView.Adapter<BrandsAdapter.Holder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
        val itemView = RowBandItemBinding.inflate(LayoutInflater.from(context), parent, false)
        return Holder(itemView)
    }

    override fun onBindViewHolder(holder: Holder, position: Int) {
        val brandModel = list?.get(position)
        var imageUrl: String? = ""
        imageUrl = if (UtilityApp.getLanguage() == Constants.English) {
            brandModel?.image
        } else {
            brandModel?.image2
        }
        try {
            GlideImg(
                context, imageUrl, R.drawable.holder_image, holder.binding.ivCatImage
            )
        } catch (e: Exception) {
            e.printStackTrace()
        }
        holder.binding.container.setOnClickListener { v -> onBrandClick.onBrandClicked(position, brandModel) }
    }

    fun setCategoriesList(list: MutableList<BrandModel>) {
        this.list = list
        notifyDataSetChanged()
    }

    override fun getItemCount(): Int {
        return if (limit != 0) list?.size?.coerceAtMost(limit)?:0 else list?.size?:0
    }

    interface OnBrandClick {
        fun onBrandClicked(position: Int, brandModel: BrandModel?)
    }

    class Holder(var binding: RowBandItemBinding) : RecyclerView.ViewHolder(
        binding.root
    )
}
